### PR TITLE
2.0.2 Release

### LINKED ISSUE #53: logoutEmail() should probably return a promise
Right now it returns `void`, but clearly there's a network request happening. Chrome is showing me an uncaught exception when there's an error with the call. It would be nice to be able to catch this in my own code.

CleanShot 2021-11-09 at 13 56 22

### PR DESCRIPTION
**Pull request by rgomezp**
## Update package
* Fixes functions that weren't returning promises.
* Convert file to TS, compiled via `prepare` npm publish lifecycle function

Fixes #53

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,13 +4,11 @@
   "description": "React OneSignal Module: Make it easy to integrate OneSignal with your React App!",
   "author": "rgomezp",
   "contributors": [{ "name": "Rodrigo Gomez-Palacio" }, { "name": "Pedro Bini" }],
-  "homepage": "https://onesignal.com",
-  "repository": "https://github.com/OneSignal/react-onesignal.git",
```
Comment (jkasten2) [inline package.json:8]: Why did we remove these two lines?

File: package.json [hunk 2]
```diff
@@ -4,13 +4,11 @@
   "description": "React OneSignal Module: Make it easy to integrate OneSignal with your React App!",
   "author": "rgomezp",
   "contributors": [{ "name": "Rodrigo Gomez-Palacio" }, { "name": "Pedro Bini" }],
-  "homepage": "https://onesignal.com",
```
Comment (jkasten2) [inline package.json:7]: PR says 2.0.2 but this file still has `"version": "2.0.1",`

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (jkasten2) [inline tsconfig.json:1]: Where did this file come from? Or how was it generated?

File: yarn.lock [hunk 4]
```diff
@@ -2,84 +2,84 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.14.5":
-  version "7.14.5"
-  resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.14.5.tgz#23b08d740e83f49c5e59945fbf1b43e80bbf4edb"
-  integrity sha512-9pzDqyc6OLDaqe+zbACgFkb6fKMNG6CObKpnYXChRsvYGyEdc7CA2BaqeOM+vOtCS5ndmJicPJhKAwYRI6UfFw==
+"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.14.5", "@babel/code-frame@^7.15.8":
+  version "7.15.8"
```
Comment (jkasten2) [inline yarn.lock:6]: What operation was run to update this lock file?

**Review by jkasten2 — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 3 above)
Comment (rgomezp) [reply tsconfig.json:1]: Came from [here](https://github.com/OneSignal/react-onesignal/blob/34ef57e4002795dd4cbfbf869c3880fec15ad28e/tsconfig.json) (the original version before the 2 release).

**Pull request closed without merging**